src/internal/m365/graph/errors.go

```diff
@@ -259,6 +260,10 @@ func IsErrUsersCannotBeResolved(err error) bool {
 	return hasErrorCode(err, noResolvedUsers) || hasErrorMessage(err, usersCannotBeResolved)
 }
 
+func IsErrSiteCouldNotBeFound(err error) bool {
```

Comment:
This, on the other hand, can be more simple than the error message, to convey the core intent.
```suggestion
func IsErrSiteNotFound(err error) bool {
```